cetmix_tower_server/models/cx_tower_reference_mixin.py

```diff
@@ -242,3 +250,71 @@ def _get_id_by_reference(self, reference):
 
         # This is in case some models will remove reference uniqueness constraint
         return records and records[0].id
+
+    @api.model
+    def _prepare_references(self, model, key_name, vals_list):
+        """
+        Prepare a dictionary of references for given model records.
+
+        This function extracts unique IDs from a list of dictionaries (vals_list)
+        based on a specified key (key_name), fetches the corresponding records
+        from the specified model, and returns a dictionary mapping record IDs to
+        their references.
+
+        Args:
+            model (str): The name of the model to fetch records from.
+            key_name (str): The key in the dictionaries of vals_list that contains
+                            the record IDs.
+            vals_list (list of dict): A list of dictionaries containing the values
+                                    to be processed.
+
+        Returns:
+            dict: A dictionary mapping record IDs to their references.
+        """
+        try:
+            CxModel = self.env[model]
+        except KeyError as err:
+            raise ValueError(
+                f"Model '{model}' does not exist. Please provide a valid model name."
+            ) from err
+
+        # Extract all unique ids from vals_list
+        line_ids = {vals.get(key_name) for vals in vals_list if vals.get(key_name)}
+
+        # Fetch all line references in a single query
+        lines = CxModel.browse(line_ids)
+        return {line.id: line.reference for line in lines}
+
+    @api.model
+    def _populate_references(self, model_name, field_name, vals_list, suffix=""):
+        """
+        Common method to populate references based on a specific field.
```

Comment:
I still cannot understand what does this function do. 
Description should be more specific. Eg `Create record references in the vals list used for 'create' method. Use references of the related model as a basis`.  Or something like that.

Reply:
okay,I'll fix it